### PR TITLE
Reduce size of images

### PR DESCRIPTION
**Pull request by Ocarthon**
Hey everyone, i noticed the size of the images and decided to check if we can reduce it a bit

## TLDR

| Image                                   | Size Before | Size After |
|-----------------------------------------|-------------|------------|
| `dockware/dev:6.6.9.0`                  | 4.87GB      | 2.23GB     |
| `dockware/essentials:latest`            | 3.76GB      | 2.29GB     |
| `dockware/essentials:shopware-6.6` (New)  | -           | 1.69GB     |

## Optimization 1 - Squash Layers
The Dockerfiles for the single images contain lots of layers, that store lots of redundant and not required data.

By adding a separate build stage and just copying over all files from the build stage, we can squash everything into a single layer. The only downside is that all `ENV` directives have to be set in both build stages. I added blocks around all those directives and duplicate them for the final stage

Size of `dockware/dev:6.6.9.0` afterwards: 3.12GB

## Optimization 2 - Do not install recommended packages
`apt-get install` by default also installs recommended packages that also contribute a lot to the final image size. I added the `-no-install-recommends` flag to all `apt-get install` commands and added some required packages.

After all packages are installed, the apt cache is cleared (i.e. the lists created by `apt-get update`).

Size of `dockware/dev:6.6.9.0` afterwards: 2.78GB

## Optimization 3 - Only install required node versions for Shopware 6.6+ and clear the nvm cache
Shopware 6.6 is only compatible with Node 20, so there is no need to install all Node versions below that. I assume this was also the wanted behavior, as the `variables.json` specifically contained only Node 20, but the other node versions have a default value of `true`.

After node is installed, the nvm cache is cleared. This cache contained the packaged node binaries.

Size of `dockware/dev:6.6.9.0` afterwards: 2.23GB

## Optimization 4 - Add essentials image for Shopware 6.6
In our use case, we always map `/var/www/html` with the project directory we are working on. We therefore do not need the ~400MB of the preinstalled shopware. The current `dockware/essentials:latest` would be ideal for this, but this images has all node versions installed. If you take a look at the TLDR at the top, this makes it slightly larger than the `dockware/dev` image.

I added a `dockware/essentials:shopware-6-6` image, that contains all the required things to run shopware 6.6 (i.e. only node 20).

For our use case, this brings the image down to 1.69GB.

## Add ext-opentelemetry

While this is not an optimization, shopware focuses OpenTelemetry for tracing and therefore makes sense to also includes this extension by default.

### CODE REVIEW

**Comment by boxblinkracer** [comment]
Hi

thank for this great PR

ill check it out

I would also like to say that there is something (big) going on in the background
im working on things like this since a while

i still try to check out if we can use it in the meantime  :)

so THANKS

**Comment by boxblinkracer** [comment]
Hi there
thank you for your efforts on this

just to let you know
we are before the release of a brand new dockware....ill see if this makes sense / works
but there will be something new ;)

**Comment by Ocarthon** [comment]
@boxblinkracer All good :) 

I am currently building my fork to test the time improvement when pulling the image from githubs docker repository in a github workflow.
